test.sh

```
@@ -0,0 +1,305 @@
+                echo -e "${YELLOW}Error: $message${NC}"
+            fi
+            ((ERROR_TESTS++))
+            ;;
+    esac
+    ((TOTAL_TESTS++))
+}
+
+# Function to run a single test file
+run_test_file() {
+    local test_file=$1
+    local test_name=$(basename "$test_file" .lua)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)